bloggies_backend/src/routes/stripe.ts

```diff
@@ -1,27 +1,140 @@
 import Checkout from '../models/stripe';
-import express, {Request, Response, NextFunction} from "express";
-// import { ensureLoggedIn } from "../middleware/auth";
-
+import express, { Request, Response, NextFunction } from "express";
+import Stripe from 'stripe';
+import User from '../models/user';
+import { ensureLoggedIn } from "../middleware/auth";
+import { MY_STRIPE_API_KEY } from '../config';
+import ExpressError from '../expressError';
 export const stripeRouter = express.Router();
 
+/* create new Stripe instance to facilitate interactions with Stripe API*/
+export const stripe = new Stripe(MY_STRIPE_API_KEY as string, {
+  apiVersion: "2020-08-27",
+});
+
+/** GET handles events that occur over a stripe session via webhook */
+stripeRouter.post("/webhook", async function (req: Request, res: Response, next: NextFunction) {
+  let event = req.body;
+  let data: any;
+
+  switch (event.type) {
+    case 'invoice.paid':
+      data = event.data.object;
+      console.log(`
+          invoice PAID for: ${data.customer}, 
+          email: ${data.customer_email}. 
+          subscription: ${data.subscription}`);
+
+      await User.startSubscription(data.subscription);
+      break;
+    case 'invoice.payment_failed':
+      data = event.data.object;
+      console.log(`
```

Comment:
Is there a reason you have these console logs here? was it for testing purposes?

Reply:
Yea, I wanted to log the stripe events when they happen in the terminal! I think it'd be helpful to know what events has happened through console logs in this endpoint throughout development.